flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/IncrementalSSSPExample.java

```diff
@@ -0,0 +1,304 @@
+				fileOutput = true;
+				verticesInputPath = args[0];
+				edgesInputPath = args[1];
+				edgesInSSSPInputPath = args[2];
+				edgeToBeRemoved = args[3];
+				outputPath = args[4];
+				maxIterations = Integer.parseInt(args[5]);
+			} else {
+				System.out.println("Executing IncrementalSSSP example with default parameters and built-in default data.");
+				System.out.println("Provide parameters to read input data from files.");
+				System.out.println("See the documentation for the correct format of input files.");
+				System.out.println("Usage: IncrementalSSSP <vertex path> <edge path> <edges in SSSP> <edge to be removed> <output path> <max iterations>");
+
+				return false;
+			}
+		}
+		return true;
+	}
+
+	private static DataSet<Vertex<Long, Double>> getVerticesDataSet(ExecutionEnvironment env) {
+		if (fileOutput) {
+			return env.readCsvFile(verticesInputPath)
+					.lineDelimiter("\n")
+					.types(Long.class, Double.class)
+					.map(new MapFunction<Tuple2<Long, Double>, Vertex<Long, Double>>() {
+
+						@Override
+						public Vertex<Long, Double> map(Tuple2<Long, Double> tuple2) throws Exception {
+							return new Vertex<Long, Double>(tuple2.f0, tuple2.f1);
+						}
+					});
+		} else {
+			System.err.println("Usage: IncrementalSSSP <vertex path> <edge path> <edges in SSSP> <edge to be removed> " +
+					"<output path> <max iterations>");
+			return IncrementalSSSPData.getDefaultVertexDataSet(env);
+		}
+	}
+
+	private static DataSet<Edge<Long, Double>> getEdgesDataSet(ExecutionEnvironment env) {
+		if (fileOutput) {
+			return env.readCsvFile(edgesInputPath)
+					.lineDelimiter("\n")
+					.types(Long.class, Long.class, Double.class)
+					.map(new MapFunction<Tuple3<Long, Long, Double>, Edge<Long, Double>>() {
+
+						@Override
+						public Edge<Long, Double> map(Tuple3<Long, Long, Double> tuple3) throws Exception {
+							return new Edge(tuple3.f0, tuple3.f1, tuple3.f2);
+						}
+					});
+		} else {
+			System.err.println("Usage: IncrementalSSSP <vertex path> <edge path> <edges in SSSP> <edge to be removed> " +
+					"<output path> <max iterations>");
+			return IncrementalSSSPData.getDefaultEdgeDataSet(env);
+		}
+	}
+
+	private static DataSet<Edge<Long, Double>> getEdgesinSSSPDataSet(ExecutionEnvironment env) {
+		if (fileOutput) {
+			return env.readCsvFile(edgesInSSSPInputPath)
+					.lineDelimiter("\n")
+					.types(Long.class, Long.class, Double.class)
+					.map(new MapFunction<Tuple3<Long, Long, Double>, Edge<Long, Double>>() {
+
+						@Override
+						public Edge<Long, Double> map(Tuple3<Long, Long, Double> tuple3) throws Exception {
+							return new Edge(tuple3.f0, tuple3.f1, tuple3.f2);
+						}
+					});
+		} else {
+			System.err.println("Usage: IncrementalSSSP <vertex path> <edge path> <edges in SSSP> <edge to be removed> " +
+					"<output path> <max iterations>");
+			return IncrementalSSSPData.getDefaultEdgesInSSSP(env);
+		}
+	}
+
+	private static Edge<Long, Double> getEdgeToBeRemoved() {
```

Comment:
how about we give this parameter as a srcId and a trgId (of the edge) instead of giving a file path for a single pair of ids?